File: dbflow-tests/src/androidTest/java/com/raizlabs/android/dbflow/test/typeconverter/CustomBooleanConverter.java
package com.raizlabs.android.dbflow.test.typeconverter;

import com.raizlabs.android.dbflow.converter.TypeConverter;

/**
 * Description: Example of a non-standard typeconverter thats used locally.
 */
public class CustomBooleanConverter extends TypeConverter<String, Boolean> {

    @Override
    public String getDBValue(Boolean model) {
        return model == null ? "" : model ? "1" : "0";
    }

    @Override
    public Boolean getModelValue(String data) {
        return (data == null || data.length() == 0) ? null : Boolean.getBoolean(data);
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/typeconverter/CustomBooleanConverter.kt
package com.raizlabs.android.dbflow.test.typeconverter

import com.raizlabs.android.dbflow.converter.TypeConverter

/**
 * Description: Example of a non-standard typeconverter thats used locally.
 */
class CustomBooleanConverter : TypeConverter<String, Boolean>() {

    override fun getDBValue(model: Boolean?): String {
        return if (model == null) "" else if (model) "1" else "0"
    }

    override fun getModelValue(data: String?): Boolean? {
        return if (data == null || data.length == 0) null else java.lang.Boolean.getBoolean(data)
    }
}
